pydantic/_internal/_generate_schema.py

```diff
@@ -1461,12 +1460,12 @@ def _extract_get_pydantic_json_schema(tp: Any, schema: CoreSchema) -> GetJsonSch
     """Extract `__get_pydantic_json_schema__` from a type, handling the deprecated `__modify_schema__`"""
     js_modify_function = getattr(tp, '__get_pydantic_json_schema__', None)
 
-    if js_modify_function is None and hasattr(tp, '__modify_schema__'):
-        warnings.warn(
-            'The __modify_schema__ method is deprecated, use __get_pydantic_json_schema__ instead',
-            DeprecationWarning,
+    if hasattr(tp, '__modify_schema__'):
```

Comment:
@adriangb I've removed, as we talked. jfyk

Reply:
@Kludex I'm wondering if maybe it's a good idea to allow both being defined and if so ignore `__modify_schema__` — that way a type can maybe try to be compatible with both pydantic v1 and pydantic v2 at the same time. If that doesn't make sense then maybe this change is fine.

Also, originally we wanted to keep support for `__modify_schema__` and just deprecate it to make it easier to migrate. Did we change our mind on that?

Reply:
@dmontagu I told Marcelo to just remove it since it seems like our attempt at backwards compatibility wasn’t being used. My intention was to just ignore the problem for now and circle back later to determine if we should add a backwards compatibility shim or not based on feedback in the beta. I’m open to adding that now provided we can make it work in most cases (it will definitely not work in cases like generics, which is the main argument to not try at all)